File: ProyectosCSharp/SoftProg/SoftProgModel/Ventas/OrdenVenta.cs
using SoftProgModel.GestClientes;
using SoftProgModel.RRHH;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgModel.Ventas
{
    public class OrdenVenta
    {
        private int idOrdenVenta;
        private Empleado empleado;
        private Cliente cliente;
        private BindingList<LineaOrdenVenta> lineasOrdenVenta;
        private double total;
        private DateTime fechaHora;
        private bool activa;

        public int IdOrdenVenta { get => idOrdenVenta; set => idOrdenVenta = value; }
        public Empleado Empleado { get => empleado; set => empleado = value; }
        public Cliente Cliente { get => cliente; set => cliente = value; }
        public BindingList<LineaOrdenVenta> LineasOrdenVenta { get => lineasOrdenVenta; set => lineasOrdenVenta = value; }
        public double Total { get => total; set => total = value; }
        public DateTime FechaHora { get => fechaHora; set => fechaHora = value; }
        public bool Activa { get => activa; set => activa = value; }
    }
}

File: BackJava_FrontCSharp_Ejemplos_SOAP_REST/BackJava/SoftProgModel/src/main/java/pe/edu/pucp/softprog/ventas/model/OrdenVenta.java
package pe.edu.pucp.softprog.ventas.model;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softprog.gestclientes.model.Cliente;
import pe.edu.pucp.softprog.rrhh.model.Empleado;

public class OrdenVenta {
    private int idOrdenVenta;
    private Empleado empleado;
    private Cliente cliente;
    private ArrayList<LineaOrdenVenta> lineasOrdenVenta;
    private double total;
    private Date fechaHora;
    private boolean activa;

    public OrdenVenta(){
        this.lineasOrdenVenta = new ArrayList<>();
    }
    
    public int getIdOrdenVenta() {
        return idOrdenVenta;
    }

    public void setIdOrdenVenta(int idOrdenVenta) {
        this.idOrdenVenta = idOrdenVenta;
    }

    public ArrayList<LineaOrdenVenta> getLineasOrdenVenta() {
        return lineasOrdenVenta;
    }

    public void setLineasOrdenVenta(ArrayList<LineaOrdenVenta> lineasOrdenVenta) {
        this.lineasOrdenVenta = lineasOrdenVenta;
    }

    public double getTotal() {
        return total;
    }

    public void setTotal(double total) {
        this.total = total;
    }

    public Date getFechaHora() {
        return fechaHora;
    }

    public void setFechaHora(Date fechaHora) {
        this.fechaHora = fechaHora;
    }

    public boolean isActiva() {
        return activa;
    }

    public void setActiva(boolean activa) {
        this.activa = activa;
    }

    public Empleado getEmpleado() {
        return empleado;
    }

    public void setEmpleado(Empleado empleado) {
        this.empleado = empleado;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }
    
    public void agregarLineaOrdenVenta(LineaOrdenVenta lov){
        this.lineasOrdenVenta.add(lov);
    }
    
    public void calcularTotal(){
        total = 0;
        for(LineaOrdenVenta lov : lineasOrdenVenta){
            total += lov.getSubtotal();
        }
    }
}